streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -721,6 +721,10 @@ private void handleTaskMigrated(final TaskMigratedException e) {
         subscribeConsumer();
     }
 
+    public long getCacheSize() {
+        return cacheResizeSize.get();
```

Comment:
Unfortunately I don't think we can reliably use this variable to read out the current cache size, since we erase it immediately after reading it out to resize the cache. We might need to introduce a new variable if we need the thread's cache size